frontend/src/store/actions/companies-actions.ts

```diff
@@ -0,0 +1,141 @@
+import { ThunkAction } from 'redux-thunk';
+import { Company } from '../../types/company';
+import { RootStateType } from '../../types/store';
+import { get, patch, post, remove } from './api';
+
+export const CREATE_NEW_COMPANY = 'CREATE_NEW_COMPANY';
+export const CREATE_NEW_COMPANY_BY_ORGNR = 'CREATE_NEW_COMPANY_BY_ORGNR';
+export const GET_ALL_COMPANIES = 'GET_ALL_COMPANIES';
+export const GET_COMPANY_BY_ID = 'GET_COMPANY_BY_ID';
+export const UPDATE_COMPANY = 'UPDATE_COMPANY';
+export const DELETE_COMPANY = 'DELETE_COMPANY';
+
+interface CreateCompanyAction {
+    type: typeof CREATE_NEW_COMPANY;
+    company: Company;
+}
+
+interface CreateCompanyByOrgnrAction {
+    type: typeof CREATE_NEW_COMPANY_BY_ORGNR;
+    company: Company;
+}
+
+interface GetCompaniesAction {
+    type: typeof GET_ALL_COMPANIES;
+    companies: Company[];
+}
+
+interface GetCompanyAction {
+    type: typeof GET_COMPANY_BY_ID;
+    company: Company;
+}
+
+interface UpdateCompanyAction {
+    type: typeof UPDATE_COMPANY;
+    id: string;
+    company: Company;
+}
+
+interface DeleteCompanyAction {
+    type: typeof DELETE_COMPANY;
+    id: string;
+}
+
+export type CompanyActionType = 
+    CreateCompanyAction |
+    CreateCompanyByOrgnrAction |
+    GetCompaniesAction |
+    GetCompanyAction |
+    UpdateCompanyAction |
+    DeleteCompanyAction;
+
+const createCompanySuccess = (company: Company): CreateCompanyAction => ({
+    type: CREATE_NEW_COMPANY,
+    company
+});
+
+const getCompaniesSuccess = (companies: Company[]): GetCompaniesAction => ({
+    type: GET_ALL_COMPANIES,
+    companies
+});
+
+const getCompanySuccess = (company: Company): GetCompanyAction => ({
+    type: GET_COMPANY_BY_ID,
+    company
+});
+
+const updateCompanySuccess = (id: string, company: Company): UpdateCompanyAction => ({
+    type: UPDATE_COMPANY,
+    id,
+    company
+});
+
+const deleteCompanySuccess = (id: string): DeleteCompanyAction => ({
+    type: DELETE_COMPANY,
+    id
+});
+
+export const createNewCompany = (company: Company): ThunkAction<Promise<Company>, RootStateType, undefined, CompanyActionType> => async (dispatch) => {

```

Comment:
Vi kan vurdere å ta ut retur-typen her til en egen type-konstant (siden den er lik for alle actions, ish):

type AppThunk<ReturnType = void> = ThunkAction<ReturnType, RootState, undefined, CompanyActionType>

eller

type AppThunk<T> = ThunkAction<T, RootState, undefined, CompanyActionType>